improver_tests/utilities/test_threshold_interpolation.py

```diff
@@ -0,0 +1,149 @@
+    """Set up a masked cube which is consistent for every threshold."""
+    data = np.array(
+        [
+            [[1.0, 0.9, 1.0], [0.8, 0.9, 0.5], [0.5, 0.2, 0.0]],
+            [[1.0, 0.5, 1.0], [0.5, 0.5, 0.3], [0.2, 0.0, 0.0]],
+            [[1.0, 0.2, 0.5], [0.2, 0.0, 0.1], [0.0, 0.0, 0.0]],
+        ],
+        dtype=np.float32,
+    )
+    mask = np.array(
+        [
+            [[0.0, 0.0, 1.0], [0.0, 1.0, 0.0], [1.0, 0.0, 0.0]],
+            [[0.0, 0.0, 1.0], [0.0, 1.0, 0.0], [1.0, 0.0, 0.0]],
+            [[0.0, 0.0, 1.0], [0.0, 1.0, 0.0], [1.0, 0.0, 0.0]],
+        ],
+        dtype=np.int8,
+    )
+
+    masked_data = np.ma.masked_array(data, mask=mask)
+
+    masked_cube = set_up_probability_cube(
+        masked_data,
+        thresholds=[100, 200, 300],
+        variable_name="visibility_in_air",
+        threshold_units="m",
+        spp__relative_to_threshold="less_than",
+    )
+    return masked_cube
+
+
+def test_basic(input_cube):
+    """Test that the plugin returns an Iris.cube.Cube with suitable units."""
+    thresholds = [100, 150, 200, 250, 300]
+    result = ThresholdInterpolation(thresholds)(input_cube)
+    assert result, Cube
+    assert result.units == input_cube.units
+
+
+def test_empty_threshold_list():
+    """
+    Test that a ValueError is raised if the threshold list is empty.
+    """
+    with pytest.raises(ValueError, match="The thresholds list cannot be empty."):
+        ThresholdInterpolation([])
+
+
+def test_metadata_copy(input_cube):
+    """
+    Test that the metadata dictionaries within the input cube, are
+    also present on the output cube.
+    """
+    input_cube.attributes = {"source": "ukv"}
+    thresholds = [100, 150, 200, 250, 300]
+    result = ThresholdInterpolation(thresholds)(input_cube)
+    assert input_cube.metadata._asdict() == result.metadata._asdict()
+
+
+def test_thresholds_different_mask(masked_cube):
+    """
+    Testing that a value error message is raised if masks are different across thresholds.
+    """
+    masked_cube.data.mask[0, 0, 0] = True
+    thresholds = [100, 150, 200, 250, 300]
+    error_msg = "The mask is expected to be constant across different slices of the"
+
+    with pytest.raises(ValueError, match=error_msg):
+        ThresholdInterpolation(thresholds)(masked_cube)
+
+
+def test_masked_cube(masked_cube):
+    """
+    Testing that a Cube is returned when inputting a masked cube.
+    """
+    thresholds = [100, 150, 200, 250, 300]
+    result = ThresholdInterpolation(thresholds)(masked_cube)
+    assert isinstance(result, Cube)
+
+
+def test_mask_consistency(masked_cube):
+    """
+    Test that the mask is the same before and after ThresholdInterpolation.
+    """
+    thresholds = [100, 150, 200, 250, 300]
+    original_mask = masked_cube.data.mask
+    result = ThresholdInterpolation(thresholds)(masked_cube).data.mask
+    np.testing.assert_array_equal(original_mask[0], result[0])
+
+
+def test_interpolated_values(input_cube):
```

Comment:
Can you please apply exactly the same parameterisation to this test as we have to the test that a cube is returned. This will then check that the method behaves as expected when handling a masked cube.

Note that numpy.testing.assert_array_equal just ignores the masked points in the comparison, which is terrifying, but for our purposes means that this test will pass for both the masked and unmasked cubes.

If you want to read more people being scared that two arrays which have completely different masking compare as true, see https://github.com/numpy/numpy/issues/14624